### ISSUE TITLE
[BUG]Block When exec df2.head

### ISSUE THREAD

**Issue by wuyeguo** [mars-project/mars#2768]
**Describe the bug**
*Sorry Blocked By df2.head(5)*
Script：https://docs.pymars.org/zh_CN/latest/installation/ray.html

```
import ray
ray.init(address="ray://172.16.210.22:10001")

import mars
import mars.tensor as mt
import mars.dataframe as md
session = mars.new_ray_session(worker_num=2, worker_mem=2 * 1024 ** 3)
mt.random.RandomState(0).rand(1000_0000, 5).sum().execute()
df = md.DataFrame(
    mt.random.rand(1000_0000, 4, chunk_size=500_0000),
    columns=list('abcd'))
print("start df.sum")
print(df.sum().execute())
print("start df.describe")
print(df.describe().execute())
# Convert mars dataframe to ray dataset
print("start to_ray_dataset")
ds = md.to_ray_dataset(df)
print(ds.schema(), ds.count())
print("start ds.filter")
ds.filter(lambda row: row["a"] > 0.5).show(5)
# Convert ray dataset to mars dataframe
print("start md.read_ray_dataset")
df2 = md.read_ray_dataset(ds)
print("start df2.head")
print(df2.head(5).execute())

```

**To Reproduce**
To help us reproducing this bug, please provide information below:
1. Your Python version：3.7.5
2. The version of Mars you use：0.8.2
3. Versions of crucial packages, such as numpy, scipy and pandas
    1.  Ray：1.9.2
    2. Numpy：1.21.5
    3. Pandas：1.3.5
    4. Scipy：1.7.3
5. Full stack of the error.
6. Minimized code to reproduce the error.

**Expected behavior**

**Additional context**
Output:
```
[ray@ml-test ~]$ python3 test_mars.py
The remote function 'ray.util.client.common.get_class_info' has been exported 100 times. It's possible that this warning is accidental, but this may indicate that the same remote function is being defined repeatedly from within many tasks and exported to all of the workers. This can be a performance issue and can be resolved by defining the remote function on the driver instead. See https://github.com/ray-project/ray/issues/6240 for more discussion.
Web service started at http://0.0.0.0:49007
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 100.0/100 [00:00<00:00, 156.03it/s]
start df.sum
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 100.0/100 [00:00<00:00, 120.31it/s]
a    4.999169e+06
b    4.999923e+06
c    4.999475e+06
d    5.000890e+06
dtype: float64
start df.describe
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 100.0/100 [00:13<00:00,  7.16it/s]
                  a             b             c             d
count  1.000000e+07  1.000000e+07  1.000000e+07  1.000000e+07
mean   4.999169e-01  4.999923e-01  4.999475e-01  5.000890e-01
std    2.886632e-01  2.886333e-01  2.886754e-01  2.886743e-01
min    9.919484e-08  6.832216e-08  6.025447e-08  3.897334e-10
25%    2.498717e-01  2.500875e-01  2.499015e-01  2.501594e-01
50%    4.999173e-01  4.999582e-01  4.998740e-01  5.001442e-01
75%    7.498571e-01  7.499878e-01  7.500200e-01  7.500296e-01
max    9.999998e-01  1.000000e+00  9.999999e-01  9.999999e-01
start to_ray_dataset
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 100.0/100 [00:00<00:00, 164.28it/s]
a: double
b: double
c: double
d: double
-- schema metadata --
pandas: '{"index_columns": [{"kind": "range", "name": null, "start": 0, "' + 676 10000000
start ds.filter
Map Progress:   0%|                                                                                                                                                                    | 0/2 [00:00<?, ?it/s]


Map Progress: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [01:52<00:00, 56.31s/it]
{'a': 0.9135708786168139, 'b': 0.8440060222343299, 'c': 0.8997797535511054, 'd': 0.636812260113353}
{'a': 0.8111735922426506, 'b': 0.7349742255152801, 'c': 0.0545104640311912, 'd': 0.8807744293773943}
{'a': 0.6509971501372513, 'b': 0.9760894305787635, 'c': 0.037421022585488184, 'd': 0.967949764430637}
{'a': 0.5236421731395025, 'b': 0.26958871922681793, 'c': 0.45709635536982207, 'd': 0.6666297475834149}
{'a': 0.8743533857339559, 'b': 0.1794161392233996, 'c': 0.1893300102139226, 'd': 0.07298901676376668}
start md.read_ray_dataset
start df2.head
  0%|                                                                                                                                                                                | 0/100 [00:00<?, ?it/s](RaySubPool pid=27251) 2022-03-01 15:03:17,855  WARNING worker.py:1690 -- Using blocking ray.get inside async actor. This blocks the event loop. Please use `await` on object ref with asyncio.gather if you want to yield execution to the event loop instead.
  0%|                                                       

```

**Comment by qinxuye** [comment]
Thanks for your report, could you take a look at this issue, is it related to ray client? @chaokunyang

**Comment by wuyeguo** [comment]
Ray Deploy In Two Servers:
···
[ray@al-bj-ml-prd ~]$ ray status
======== Autoscaler status: 2022-03-01 14:54:03.979654 ========
Node status
---------------------------------------------------------------
Healthy:
 2 local.cluster.node
Pending:
 (no pending nodes)
Recent failures:
 (no failures)

Resources
---------------------------------------------------------------
Usage:
 4.0/8.0 CPU (4.0 used of 4.0 reserved in placement groups)
 0.00/13.457 GiB memory
 0.97/6.204 GiB object_store_memory

Demands:
 (no resource demands)

···

**Comment by wuyeguo** [comment]
Thinks for quickly fix：
I update my mars with
pip3 install git+https://github.com/mars-project/mars@master

when run my code will get：
Ray Version：1.9.2
is there a suggested version of ray?

```
(RayMainPool pid=None, ip=172.16.210.21) 2022-03-02 11:29:53,700        ERROR serialization.py:283 -- 'pickle'
(RayMainPool pid=None, ip=172.16.210.21) Traceback (most recent call last):
(RayMainPool pid=None, ip=172.16.210.21)   File "/usr/local/python3/lib/python3.7/site-packages/ray/serialization.py", line 281, in deserialize_objects
(RayMainPool pid=None, ip=172.16.210.21)     obj = self._deserialize_object(data, metadata, object_ref)
(RayMainPool pid=None, ip=172.16.210.21)   File "/usr/local/python3/lib/python3.7/site-packages/mars/oscar/backends/ray/communication.py", line 78, in _deserialize_object
(RayMainPool pid=None, ip=172.16.210.21)     message = deserialize(*value.message)
(RayMainPool pid=None, ip=172.16.210.21)   File "/usr/local/python3/lib/python3.7/site-packages/mars/serialization/core.py", line 423, in deserialize
(RayMainPool pid=None, ip=172.16.210.21)     gen_deserialized = _deserialize(*gen_to_deserial)
(RayMainPool pid=None, ip=172.16.210.21)   File "/usr/local/python3/lib/python3.7/site-packages/mars/serialization/core.py", line 397, in _deserialize
(RayMainPool pid=None, ip=172.16.210.21)     serializer = _deserializers[serializer_name]
(RayMainPool pid=None, ip=172.16.210.21) KeyError: 'pickle'
Traceback (most recent call last):
  File "/usr/local/python3/lib/python3.7/site-packages/mars/deploy/oscar/ray.py", line 320, in new_cluster
    await cluster.start()
  File "/usr/local/python3/lib/python3.7/site-packages/mars/deploy/oscar/ray.py", line 481, in start
    NodeRole.WORKER, self.supervisor_address, self.supervisor_address
  File "/usr/local/python3/lib/python3.7/site-packages/mars/deploy/oscar/ray.py", line 95, in create
    address=lookup_address,
  File "/usr/local/python3/lib/python3.7/site-packages/mars/oscar/api.py", line 27, in create_actor
    return await ctx.create_actor(actor_cls, *args, uid=uid, address=address, **kwargs)
  File "/usr/local/python3/lib/python3.7/site-packages/mars/oscar/backends/context.py", line 105, in create_actor
    result = await self._wait(future, address, create_actor_message)
  File "/usr/local/python3/lib/python3.7/site-packages/mars/oscar/backends/context.py", line 83, in _wait
    return await future
  File "/usr/local/python3/lib/python3.7/site-packages/mars/oscar/backends/context.py", line 74, in _wait
    await asyncio.shield(future)
  File "/usr/local/python3/lib/python3.7/site-packages/mars/oscar/backends/core.py", line 50, in _listen
    message: _MessageBase = await client.recv()
  File "/usr/local/python3/lib/python3.7/site-packages/mars/oscar/backends/communication/base.py", line 262, in recv
    return await self.channel.recv()
  File "/usr/local/python3/lib/python3.7/site-packages/mars/oscar/backends/ray/communication.py", line 196, in recv
    result = await object_ref
ray.exceptions.RayTaskError: ray::RayMainPool.__on_ray_recv__() (pid=28605, ip=172.16.210.21, repr=<mars.oscar.backends.ray.pool.RayMainPool object at 0x7ff1b06cc550>)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RaySystemError: System error: 'pickle'
traceback: Traceback (most recent call last):
  File "/usr/local/python3/lib/python3.7/site-packages/ray/serialization.py", line 281, in deserialize_objects
    obj = self._deserialize_object(data, metadata, object_ref)
  File "/usr/local/python3/lib/python3.7/site-packages/mars/oscar/backends/ray/communication.py", line 78, in _deserialize_object
    message = deserialize(*value.message)
  File "/usr/local/python3/lib/python3.7/site-packages/mars/serialization/core.py", line 423, in deserialize
    gen_deserialized = _deserialize(*gen_to_deserial)
  File "/usr/local/python3/lib/python3.7/site-packages/mars/serialization/core.py", line 397, in _deserialize
    serializer = _deserializers[serializer_name]
KeyError: 'pickle'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "test_mars.py", line 7, in <module>
    session = mars.new_ray_session(worker_num=2, worker_mem=2 * 1024 ** 3)
  File "/usr/local/python3/lib/python3.7/site-packages/mars/deploy/oscar/ray.py", line 363, in new_ray_session
    client = new_cluster_in_ray(**new_cluster_kwargs)
  File "/usr/local/python3/lib/python3.7/site-packages/mars/deploy/oscar/ray.py", line 335, in new_cluster_in_ray
    client = fut.result()
  File "/usr/local/python3/lib/python3.7/concurrent/futures/_base.py", line 435, in result
    return self.__get_result()
  File "/usr/local/python3/lib/python3.7/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
  File "/usr/local/python3/lib/python3.7/site-packages/mars/deploy/oscar/ray.py", line 327, in new_cluster
    raise stop_ex from ex
  File "/usr/local/python3/lib/python3.7/site-packages/mars/deploy/oscar/ray.py", line 325, in new_cluster
    await cluster.stop()
  File "/usr/local/python3/lib/python3.7/site-packages/mars/deploy/oscar/ray.py", line 519, in stop
    await stop_supervisor(self.supervisor_address, self._config)
  File "/usr/local/python3/lib/python3.7/site-packages/mars/deploy/oscar/service.py", line 66, in stop_supervisor
    await stop_services(NodeRole.SUPERVISOR, address=address, config=config)
  File "/usr/local/python3/lib/python3.7/site-packages/mars/services/core.py", line 185, in stop_services
    await asyncio.gather(*[inst.stop() for inst in instances])
  File "/usr/local/python3/lib/python3.7/site-packages/mars/services/task/supervisor/service.py", line 52, in stop
    uid=TaskConfigurationActor.default_uid(), address=self._address
  File "/usr/local/python3/lib/python3.7/site-packages/mars/oscar/api.py", line 37, in destroy_actor
    return await ctx.destroy_actor(actor_ref)
  File "/usr/local/python3/lib/python3.7/site-packages/mars/oscar/backends/context.py", line 121, in destroy_actor
    result = await self._wait(future, actor_ref.address, message)
  File "/usr/local/python3/lib/python3.7/site-packages/mars/oscar/backends/context.py", line 83, in _wait
    return await future
  File "/usr/local/python3/lib/python3.7/site-packages/mars/oscar/backends/context.py", line 74, in _wait
    await asyncio.shield(future)
  File "/usr/local/python3/lib/python3.7/site-packages/mars/oscar/backends/core.py", line 50, in _listen
    message: _MessageBase = await client.recv()
  File "/usr/local/python3/lib/python3.7/site-packages/mars/oscar/backends/communication/base.py", line 262, in recv
    return await self.channel.recv()
  File "/usr/local/python3/lib/python3.7/site-packages/mars/oscar/backends/ray/communication.py", line 196, in recv
    result = await object_ref
ray.exceptions.RayTaskError: ray::RayMainPool.__on_ray_recv__() (pid=28605, ip=172.16.210.21, repr=<mars.oscar.backends.ray.pool.RayMainPool object at 0x7ff1b06cc550>)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RaySystemError: System error: 'pickle'
traceback: Traceback (most recent call last):
  File "/usr/local/python3/lib/python3.7/site-packages/ray/serialization.py", line 281, in deserialize_objects
    obj = self._deserialize_object(data, metadata, object_ref)
  File "/usr/local/python3/lib/python3.7/site-packages/mars/oscar/backends/ray/communication.py", line 78, in _deserialize_object
    message = deserialize(*value.message)
  File "/usr/local/python3/lib/python3.7/site-packages/mars/serialization/core.py", line 423, in deserialize
    gen_deserialized = _deserialize(*gen_to_deserial)
  File "/usr/local/python3/lib/python3.7/site-packages/mars/serialization/core.py", line 397, in _deserialize
    serializer = _deserializers[serializer_name]
KeyError: 'pickle'
(RayMainPool pid=None, ip=172.16.210.21) 2022-03-02 11:29:53,744        ERROR serialization.py:283 -- 'pickle'
(RayMainPool pid=None, ip=172.16.210.21) Traceback (most recent call last):
(RayMainPool pid=None, ip=172.16.210.21)   File "/usr/local/python3/lib/python3.7/site-packages/ray/serialization.py", line 281, in deserialize_objects
(RayMainPool pid=None, ip=172.16.210.21)     obj = self._deserialize_object(data, metadata, object_ref)
(RayMainPool pid=None, ip=172.16.210.21)   File "/usr/local/python3/lib/python3.7/site-packages/mars/oscar/backends/ray/communication.py", line 78, in _deserialize_object
(RayMainPool pid=None, ip=172.16.210.21)     message = deserialize(*value.message)
(RayMainPool pid=None, ip=172.16.210.21)   File "/usr/local/python3/lib/python3.7/site-packages/mars/serialization/core.py", line 423, in deserialize
(RayMainPool pid=None, ip=172.16.210.21)     gen_deserialized = _deserialize(*gen_to_deserial)
(RayMainPool pid=None, ip=172.16.210.21)   File "/usr/local/python3/lib/python3.7/site-packages/mars/serialization/core.py", line 397, in _deserialize
(RayMainPool pid=None, ip=172.16.210.21)     serializer = _deserializers[serializer_name]
(RayMainPool pid=None, ip=172.16.210.21) KeyError: 'pickle'
```

**Comment by chaokunyang** [comment]
Hi @wuyeguo , I believe this bug is fixed in #2773. Could you try the fix?

**Comment by qinxuye** [comment]
> Hi @chaokunyang ， It's OK by the second commit：https://github.com/mars-project/mars.git@4474103aa61ffbb08d62481388a3e1c7f9dba98d
> 
> my test code can success run
> 
> thank you for your quickly fix。 @chaokunyang @qinxuye

Nice. Expected to see #2773 included in 0.9.0b2 & 0.8.3.
